alibi_detect/utils/schemas.py

```diff
@@ -0,0 +1,431 @@
+# mypy: ignore-errors
+# TODO - conditional checks depending on backend
+# TODO - consider validating output of get_config calls
+
+import numpy as np
+from pydantic import BaseModel
+from typing import Optional, Union, Dict, List, Callable
+from alibi_detect.utils._types import Literal
+from alibi_detect.version import __version__, __config_spec__
+from alibi_detect.cd.tensorflow import HiddenOutput, UAE
+from alibi_detect.models.tensorflow import TransformerEmbedding
+from transformers import PreTrainedTokenizerBase
+from alibi_detect.utils.frameworks import has_pytorch, has_tensorflow  # , has_sklearn
+
+
+SUPPORTED_MODELS = Union['UAE', 'HiddenOutput', 'tf.keras.Model']
+SupportedModels = [UAE, HiddenOutput]
```

Comment:
@jklaise @mauicv Would appreciate input on this one.

`SUPPORTED_MODELS` is used for typing, wheras `SupportedModels` is a list of classes, to be used later on in `isinstance` checks i.e. https://github.com/SeldonIO/alibi-detect/blob/223b4cea07f4d767bce50f6f8ccdf065584a25c1/alibi_detect/utils/saving.py#L758. 

Ideally we'd only have one of them. However, I wasn't able to construct the typing `Union` from a list (or tuple) of classes. The reverse should be possible (i.e. use `Union` contents in `isinstance`, see first answer https://stackoverflow.com/questions/45957615/check-a-variable-against-union-type-at-runtime-in-python-3-6), but I wasn't able to get this working in our case since we use forward references (to handle TF or torch not being installed).

Reply:
Ah I see. As a minimum I would document the different uses of the two constants here.

Reply:
I was going to suggest:

```py
reduce(lambda a, b: Union[a, b], SupportedModels)
```

but your way looks like the right way to do it!

Reply:
Ooh very pythonic. Shall keep this one in mind in case the current solution doesn't work in part III. Thanks!